lib/Service/ShareService.php

```diff
@@ -44,20 +46,36 @@ class ShareService
      */
     protected $logger;
 
+    /**
+     * @var IURLGenerator
+     */
+    private $url;
+
+    /**
+     * @var IL10N
+     */
+    private $l;
+
     /**
      * ShareService constructor
      *
      * @param FeedServiceV2   $feedService Service for feeds
      * @param ItemServiceV2   $itemService Service to manage items
+     * @param IURLGenerator   $url         URL Generator
+     * @param IL10N           $l           Localization interface
      * @param LoggerInterface $logger      Logger
      */
     public function __construct(
         FeedServiceV2 $feedService,
         ItemServiceV2 $itemService,
+        IURLGenerator $url,
+        IL10N $l,
```

Comment:
please give these more descriptive names. It's not a URL. It's a URL generator.

Reply:
My bad, I was basing my implementation on existing code (in Settings/AdminSection, `IURLGenerator $url`)
I've corrected the ShareService code.

By the way, could we get an update on the unresolved reviews? We've resolved some of them, but there are a couple that are still open:
https://github.com/nextcloud/news/pull/1191#discussion_r588856418
https://github.com/nextcloud/news/pull/1191#discussion_r588855739
https://github.com/nextcloud/news/pull/1191#discussion_r588859055

Thanks!